### DISCUSSION TITLE
Why merge all sources inside a single `input` object ?

### DISCUSSION THREAD

**Discussion by HenriJ** [RobinTail/express-zod-api#2178] [General]
Hi @RobinTail ,
So nice to discover this project and how clean and thought out it is! I was also evaluating tRPC, and I feel express-zod-api is providing the same ease-of-use and type-safety while still keeping compatibility with good old rest which is great 👍  !

It was a bit surprising for me when first testing express-zod-api to see that all sources are merged inside a single input object. I was naively expecting to have for each route a inputParams, inputQuery, inputHeaders and inputBody.  

Maybe you could share the reasoning behind this design decision?

Thank you again for this great project!

**Comment by RobinTail** [comment]
Hello @HenriJ ,

> inputParams, inputQuery, inputHeaders and inputBody

Exactly for that reason.
I wanted to shift the developer focus to data itself instead of bothering about the way it's coming — in `query` or in `body`.
It also enables the level of flexibility for consumers I believe it should have.

**Reply by HenriJ** [reply]
Ok that makes sense :)

In my case I was converting a previous POST endpoint where I had both a query parameter and a body, and this behavior was a bit surprising at first (when looking at the generated swagger).

About tRPC, I guess it could bring visibility to express-zod-api to maybe add a comparison in the README. It seems to me that your project is superior on many levels and I wish I found out about it before.

**Reply by RobinTail** [reply]
Thank you for the appreciation, @HenriJ .
Yes, such a comparison can be useful for those who'd like to migrate from tRPC.
In case you feel up to making such a comparison, feel free to contribute.
